mcf-browser/src/back-end/data.ts

```diff
@@ -0,0 +1,250 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const ID_DELIMITER = ',';
+
+const ID_PROPERTIES = [
+  'variableMeasured',
+  'observationAbout',
+  'provenance',
+  'measurementMethod',
+  'observationPeriod',
+  'unit',
+  'scalingFactor',
+] as const;
+type IdProperties = typeof ID_PROPERTIES[number];
+type SeriesIdObject = Pick<Series, IdProperties>;
+
+type DataPoint = {
+  x: string;
+  y: number;
+};
+
+/** Class representation of a single series */
+class Series {
+  /** A unique identifier made using the facet
+  */
+  id: string;
+
+  /** An ordered list of datapoints */
+  data: DataPoint[];
+
+  /** The observationAbout of the data */
+  observationAbout?: string;
+
+  /** The variable being measured */
+  variableMeasured?: string;
+
+  /** The provenance of the data */
+  provenance?: string;
```

Comment:
These are required props of a Series?

Reply:
Not every tmcf/csv file has these properties filled in so I left it as optional, do you think it's better to make these properties required and then force the user to pass in null?

Reply:
Actually, provenance may not show up at this point (it gets auto-generated later), but `variableMeasured`, `observationAbout` are required.

Obs with those empty should result in warning.  The MCF Graph browser does a bunch of careful validation/error tracking, and it might be nice to rely-on/reuse that...

Reply:
I think the MCF Graph browser has error tracking for parsing issues (I couldn't find the error tracking for the presence of specific properties) but I could mimic that format to throw a parsing error that gets displayed.